react-app/src/components/ProposalDetailScreen/ProposalDescription.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useCallback } from "react";
+import MDEditor from "@uiw/react-md-editor";
+import cn from "classnames";
+import { toast } from "react-toastify";
+import Paper from "../common/Paper/Paper";
+import IconButton from "../common/Buttons/IconButton";
+import { IconType } from "../common/Icons/Icons";
+import { useClipboard } from "../../hooks/useClipboard";
+import { useLocale } from "../../providers/AppLocaleProvider";
+import { Proposal } from "../../generated/graphql";
+
+const ProposalDescription: React.FC<{ proposal: Proposal }> = ({
+  proposal,
+}) => {
+  const copy = useClipboard();
+  const { translate } = useLocale();
+
+  const shareAPIAvailable = !!navigator.share;
+
+  const handleCopyLink = useCallback(() => {
+    copy(window.location.href)
+      .then(() => {
+        toast.success(translate("UserInfoPanel.addressCopied"));
+      })
+      .catch((err) => {
+        console.error(`Failed to copy text = `, err);
+      });
```

Comment:
would `toast.promise` work for this case?

Reply:
`toast.promise` will show two toast on pending and success (or error), but we only want to show one toast on success? Since this promise resolves almost instantly